specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-01-31-preview/redhatopenshift.json

```diff
@@ -0,0 +1,903 @@
+      }
+    },
+    "EncryptionAtHost": {
+      "description": "EncryptionAtHost represents encryption at host state",
+      "enum": [
+        "Disabled",
+        "Enabled"
+      ],
+      "type": "string"
+    },
+    "IngressProfile": {
+      "description": "IngressProfile represents an ingress profile.",
+      "properties": {
+        "name": {
+          "description": "The ingress profile name.",
+          "type": "string"
+        },
+        "visibility": {
+          "$ref": "#/definitions/Visibility",
+          "description": "Ingress visibility."
+        },
+        "ip": {
+          "description": "The IP of the ingress.",
+          "type": "string"
+        }
+      }
+    },
+    "MasterProfile": {
+      "description": "MasterProfile represents a master profile.",
+      "properties": {
+        "vmSize": {
+          "$ref": "#/definitions/VMSize",
+          "description": "The size of the master VMs."
+        },
+        "subnetId": {
+          "description": "The Azure resource ID of the master subnet.",
+          "type": "string"
+        },
+        "encryptionAtHost": {
+          "$ref": "#/definitions/EncryptionAtHost",
+          "description": "Whether master virtual machines are encrypted at host."
+        },
+        "diskEncryptionSetId": {
+          "description": "The resource ID of an associated DiskEncryptionSet, if applicable.",
+          "type": "string"
+        }
+      }
+    },
+    "NetworkProfile": {
+      "description": "NetworkProfile represents a network profile.",
+      "properties": {
+        "sdnProvider": {
+          "$ref": "#/definitions/SDNProvider",
+          "description": "The SDNProvider to use when installing the cluster."
+        },
+        "podCidr": {
+          "description": "The CIDR used for OpenShift/Kubernetes Pods.",
+          "type": "string"
+        },
+        "serviceCidr": {
+          "description": "The CIDR used for OpenShift/Kubernetes Services.",
+          "type": "string"
+        }
+      }
+    },
+    "OpenShiftCluster": {
+      "description": "OpenShiftCluster represents an Azure Red Hat OpenShift cluster.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/OpenShiftClusterProperties",
+          "description": "The cluster properties.",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "The system meta data relating to this resource.",
+          "readOnly": true
+        }
+      }
+    },
+    "OpenShiftClusterAdminKubeconfig": {
+      "description": "OpenShiftClusterAdminKubeconfig represents an OpenShift cluster's admin kubeconfig.",
+      "properties": {
+        "kubeconfig": {
```

Comment:
>kubeconfig [](http://example.com/codeflow?start=9&length=10)

If this is a secret, it should be tagged with x-ms-secret: true.

Reply:
an you elaborate on how this looks like:
```
  "OpenShiftClusterAdminKubeconfig": {
      "description": "OpenShiftClusterAdminKubeconfig represents an OpenShift cluster's admin kubeconfig.",
      "type": "object",
      "properties": {
        "kubeconfig": {
          "description": "The base64-encoded kubeconfig file.",
          "type": "string",
          "x-ms-secret": true
        }
      }
    },
```

gives me on SDK generation test:
```
Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.38->2.3.38)
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
    - file:///swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/redhatopenshift.json:663:8 ($.definitions.OpenShiftClusterCredentials.properties.kubeadminPassword)
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
    - file:///swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/redhatopenshift.json:648:8 ($.definitions.OpenShiftClusterAdminKubeconfig.properties.kubeconfig)
FATAL: swagger-document/individual/schema-validator - FAILED
FATAL: Error: [OperationAbortedException] Error occurred. Exiting.
```
can it be our autorest image we use is old?

Reply:
Not sure: looks correct to me. @ArcturusZhang, any idea?